src/components/PasswordComplexity.tsx

```diff
@@ -0,0 +1,74 @@
+import styled from 'styled-components';
+import { openSans } from '@/styles/fonts';
+import COLORS from '@/styles/colors';
+import { Dispatch, SetStateAction } from 'react';
+import Icon from '../../assets/icons/Icon';
+
+const PasswordComplexityDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 4px;
+`;
+
+const PasswordRequirementDiv = styled.div<{ met: boolean }>`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+`;
+
+const PasswordRequirementText = styled.p<{ met: boolean }>`
+  ${openSans.style}
+  color: ${props => (props.met ? COLORS.green : COLORS.greyMid)};
+`;
+
+function PasswordRequirement({ met, text }: { met: boolean; text: string }) {
+  return (
+    <PasswordRequirementDiv met={met}>
+      <Icon type={met ? 'green_check' : 'gray_dot'} />
+      <PasswordRequirementText met={met}>{text}</PasswordRequirementText>
+    </PasswordRequirementDiv>
+  );
+}
+
+export default function PasswordComplexity({
+  password,
+  setComplexity,
+}: {
+  password: string;
+  setComplexity: Dispatch<SetStateAction<boolean>>;
+}) {
+  const hasUpperCase = /[A-Z]/.test(password);
+  const hasLowerCase = /[a-z]/.test(password);
+  const hasNumber = /\d/.test(password);
+  const hasSpecial = /[!@#$%^&*(),.?":{}|<>]/.test(password);
```

Comment:
When testing I was wondering why some special characters weren't working, and realized this might be confusing for the users. I thought we can either display what special characters are allowed or just allow any non-alphanumeric character. I'm not sure if restricting them to a subset of special characters would be the way to go though if we are limiting the user's options. 

Also, apparently the National Institute of Standards and Technology did research around password complexity, and now[ recommends against strict password complexity requirements](https://www.reddit.com/r/UXDesign/comments/167e9hj/the_nist_no_longer_recommends_password/). However, I still like the idea of having some password complexity requirements. Should we consider removing the special characters requirement? Looks like length is preferred over arbitrary special characters, so we can always increase the password length requirement for increased security. Curious what your thoughts are on this? (side note: GWN does not have a special character requirement)